network/hybridNetwork.go

```diff
@@ -206,8 +205,12 @@ func (n *HybridP2PNetwork) OnNetworkAdvance() {
 
 // GetHTTPRequestConnection returns the underlying connection for the given request. Note that the request must be the same
 // request that was provided to the http handler ( or provide a fallback Context() to that )
-func (n *HybridP2PNetwork) GetHTTPRequestConnection(request *http.Request) (conn net.Conn) {
-	return nil
+func (n *HybridP2PNetwork) GetHTTPRequestConnection(request *http.Request) (conn DeadlineSettable) {
```

Comment:
This is really where i should have asked, why not have this return an err? Because its normal state for connections to not/no longer to exist?

Reply:
because it is expected to have all connections to be tracked so network should know about them